src/m365/onenote/commands/notebook/notebook-list.ts

```diff
@@ -58,6 +66,11 @@ class OneNoteNotebookListCommand extends GraphCommand {
   #initValidators(): void {
     this.validators.push(
       async (args: CommandArgs) => {
+        const options = [args.options.userId, args.options.userName, args.options.groupId, args.options.groupName, args.options.webUrl];
+        if (options.filter(item => item !== undefined).length > 1) {
+          return `Specify either userId, userName, groupId, groupName or webUrl, but not multiple`;
+        }
```

Comment:
Let's make use of option sets here instead of a custom validation check

Reply:
We can also specify nothing, which will result in retrieving the notebooks of the current user. Using optionsets would error.

Reply:
You should be able to make use of the property `runsWhen` to setup conditional option sets.